src/lib/loadenv.js

```diff
@@ -0,0 +1,33 @@
+// @ts-check
+
+const { readFileSync, accessSync, constants } = require("fs")
+const { parse } = require("dotenv")
+
+function loadEnv(env) {
+  return checkFileExistsSync(env) ? parse(readFileSync(env)) : {}
+}
+
+function applyToEnv(config) {
+  if (Object.keys(config).length !== 0) {
+    for (const k in config) {
+      if (!process.env[k]) {
```

Comment:
Since `.env.shared` is loaded before `.env`, I expect this conditional to prevent a var `FOO` in `.env` from overriding `FOO` in `.env.shared`, since by the time when `.env` is loaded, `process.env['FOO']` already exists (was loaded from `.env.shared`). But my testing shows that the overriding does work. Javascript magic?

Reply:
I believe you are exactly right, it is JS magic. Namely a result of how the spread operator works on objects literals.

```
let obj1 = { foo: 'bar', x: 42 };
let obj2 = { foo: 'baz', y: 13 };

let clonedObj = { ...obj1 };
// Object { foo: "bar", x: 42 }

let mergedObj = { ...obj1, ...obj2 };
// Object { foo: "baz", x: 42, y: 13 }
```
[SOURCE](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals)

Reply:
@icirellik please correct me if im wrong :-)

Reply:
i see. thanks for clarifying!

Reply:
@ovasdi that's exactly right. The spread operator functions similarly to `Object.assign` where the right-most object overrides and keys assigned before it.